### DISCUSSION TITLE
Interactive regime

### DISCUSSION THREAD

**Discussion by DanRRRR** [epochpic/epoch#614] [General]
Would be nice to have in Epoch interactive regime. Say, you want the running code to immediately dump the data, or provide detailed report of its current state or pause the run, or ideally even plot something, you issue some command and it is done. Any ideas how better do that? That is how all my own codes work. May be something like that already partially present in the code?

**Comment by Status-Mirror** [comment]
Hey @DanRRRR,

A similar feature is already present in the code, but it's a bit of a hack. If you create a file called `STOP` in the same directory as the `input.deck`, then the simulation will finish the current step, write a restart dump, then stop. You could then inspect the current state, and restart the simulation.

This feature was originally developed for running on clusters where you may only have a certain time allocation. It's discussed more in the [cluster restart](https://epochpic.github.io/quickstart/basic_examples/restart_example.html#restart-at-end-of-the-cluster-job-allocation) demo.

Hope this helps,
Stuart

**Comment by DanRRRR** [comment]
Cool, thanks for this info, Stuart

**Comment by DanRRRR** [comment]
How about adding emergency printing regular data without stopping of run?

**Reply by Status-Mirror** [reply]
Hey @DanRRRR,

Good idea - I'll see if we can make it an option in the new version.

Cheers,
Stuart

**Comment by DanRRRR** [comment]
Do any hacks exist besides Restart to prevent the running code from stopping at (incorrectly) defined in input.deck  end time t_end and force it to continue ? 

So much time was lost due to absence of this option
